### DISCUSSION TITLE
Mikrotik

### DISCUSSION THREAD

**Discussion by mikrologic** [edgenative/mikrotik-irrupdater#1] [General]
I see that the current behavior is to create all the applicable rules in individual rules with the same chain. Is this better than having a single rule per chain with all of the applicable entries contained in the single rule?

**Comment by edgenative** [comment]
It wasn't clear to me what was more performant for the box to evaluate.  Some of my filters have 6000+ prefixes in them - my concern was length of the rule becoming an issue at some point in the future. 

Did you have an opinion?

**Reply by mikrologic** [reply]
I am not sure the best path either. I like the look of a single rule. perhaps a config option per AS for combined or separate? or a global config option that if any AS has prefix count greater > X generate individual rules?

**Reply by edgenative** [reply]
I need to find a way of testing the evaluation performance.  Certainly this way was easier todo the comparison.  My hope in the future was to actually extend this to only delete the changed/removed lines rather than replacing the whole filter - just haven't had cycles to play with that yet beyond finding the behaviour wasn't consistent. 

Not a bad idea of what people prefer to see.  Open to cutting a PR too :)
